doc/source/serve/core-apis.rst

```diff
@@ -1,128 +1,150 @@
 
-A backend consists of a number of *replicas*, which are individual copies of the function or class that are started in separate Ray Workers (processes).
+A deployment consists of a number of *replicas*, which are individual copies of the function or class that are started in separate Ray Actors (processes).
 
 .. code-block:: python
 
-  def handle_request(starlette_request):
-    return "hello world"
-
-  class RequestHandler:
+  @serve.deployment
+  class MyFirstDeployment:
     # Take the message to return as an argument to the constructor.
     def __init__(self, msg):
         self.msg = msg
 
-    def __call__(self, starlette_request):
+    def __call__(self, request):
+        return self.msg
+
+    def other_method(self, arg):
         return self.msg
 
-  serve.create_backend("simple_backend", handle_request)
-  # Pass in the message that the backend will return as an argument.
-  # If we call this backend, it will respond with "hello, world!".
-  serve.create_backend("simple_backend_class", RequestHandler, "hello, world!")
+  MyFirstDeployment.deploy("Hello world!")
 
-We can also list all available backends and delete them to reclaim resources.
-Note that a backend cannot be deleted while it is in use by an endpoint because then traffic to an endpoint may not be able to be handled.
+Deployments can be exposed in two ways: over HTTP or in Python via the :ref:`servehandle-api`.
+By default, HTTP requests will be forwarded to the ``__call__`` method of the class (or the function) and a ``Starlette Request`` object will be the sole argument.
+You can also define a deployment that wraps a FastAPI app for more flexible handling of HTTP requests. See :ref:`serve-fastapi-http` for details.
+
+We can also list all available deployments and dynamically get a reference to them:
 
 .. code-block:: python
 
-  >> serve.list_backends()
-  {
-      'simple_backend': {'num_replicas': 1},
-      'simple_backend_class': {'num_replicas': 1},
-  }
-  >> serve.delete_backend("simple_backend")
-  >> serve.list_backends()
-  {
-      'simple_backend_class': {'num_replicas': 1},
-  }
-
-Exposing a Backend
-==================
-
-While backends define the implementation of your request handling logic, endpoints allow you to expose them via HTTP.
-Endpoints are "logical" and can have one or multiple backends that serve requests to them.
-To create an endpoint, we simply need to specify a name for the endpoint, the name of a backend to handle requests to the endpoint, and the route and methods where it will be accesible.
-By default endpoints are serviced only by the backend provided to :mod:`create_endpoint <ray.serve.api.create_endpoint>`, but in some cases you may want to specify multiple backends for an endpoint, e.g., for A/B testing or incremental rollout.
-For information on how to do this, please see :ref:`serve-split-traffic`.
+  >> serve.list_deployments()
+  {'MyFirstDeployment': <ray.serve.api.ServeDeployment object at 0x7fe0c04c42b0>}
```

Comment:
oof, we need a better `__repr__` for ServeDeployment

Reply:
Added! There was already a __str__, I just called into it for repr